x-pack/plugin/core/src/test/java/org/elasticsearch/xpack/core/ml/job/config/AnalysisConfigTests.java

```diff
@@ -701,6 +718,7 @@ public void testVerify_GivenComplexPerPartitionCategorizationConfig() {
         AnalysisConfig.Builder analysisConfig = new AnalysisConfig.Builder(detectors);
         analysisConfig.setBucketSpan(TimeValue.timeValueHours(1));
         analysisConfig.setLatency(TimeValue.ZERO);
+        analysisConfig.setModelPruneWindow(TimeValue.ZERO);
```

Comment:
Zero shouldn't be valid, as it's null that means "don't prune after a fixed time".